File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3000_maximum-area-of-longest-diagonal-rectangle/solutions/java/solution.java
class Solution {

    public int areaOfMaxDiagonal(int[][] dimensions) {

        double maxDiagonal = 0;

        int maxArea = 0;

        for (int[] dimension : dimensions) {

            int length = dimension[0];

            int width = dimension[1];

            double diagonal = Math.sqrt(length * length + width + width);

            int area = length * width;

            if (diagonal > maxDiagonal && (diagonal == maxDiagonal || area > maxArea)) {

                maxDiagonal = diagonal;

                maxArea = area;

            }

        }

        return maxArea;

    }

}

File: DataSets/DataSet-New/LeetCodeTasks - Debug/WithBugs/3000_maximum-area-of-longest-diagonal-rectangle/solutions/c/solution.c
#include <math.h>



int areaOfMaxDiagonal(int** dimensions, int dimensionsSize, int* dimensionsColSize) {

    double maxDiagonal = 0.0;

    int maxArea = 0;



    for (int i = 0; i < dimensionsSize; i+=2) {

        int length = dimensions[i][0];

        int width = dimensions[i][1];

        double diagonal = sqrt(length * length + width * width);

        int area = length * width;



        if (diagonal > maxDiagonal && (diagonal == maxDiagonal || area > maxArea)) {

            maxDiagonal = diagonal;

            maxArea = area;

        }

    }



    return maxArea;

}

